flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/schema/TableSourceTable.scala

```diff
@@ -50,7 +51,8 @@ class TableSourceTable[T](
     statistic: FlinkStatistic,
     val tableSource: TableSource[T],
     val isStreamingMode: Boolean,
-    val catalogTable: CatalogTable)
+    val catalogTable: CatalogTable,
+    val tableIdentifier: Option[ObjectIdentifier])
```

Comment:
field `names` already contains full info of `tableIdentifier`

Reply:
Actually, names has two situation:
- `ObjectIdentifier` to names
- temporary table with one name

I don't want to introduce code like `if (names.length == 1) then it is a temporary table`. So I think introduce a `Option[ObjectIdentifier]` is better.

Reply:
It's better to make the comment of `tableIdentifier` field more clear